hadoop-ozone/dist/src/main/compose/xcompat/lib.sh

```diff
@@ -107,7 +107,162 @@ test_cross_compatibility() {
     done
   done
 
+    # NEW: Add checkpoint compatibility tests
+  # Give OM a moment to be fully ready for HTTP requests
+  echo "Waiting for OM to be ready for HTTP requests..."
+  sleep 10
+
+  echo ""
+  echo "=========================================="
+  echo "Running checkpoint compatibility tests"
+  echo "=========================================="
+  for client_version in ${checkpoint_client_versions}; do
+    client _test_checkpoint_compatibility
+  done
+
   KEEP_RUNNING=false stop_docker_env
 }
 
+_get_om_hostname() {
+  # Get OM hostname from the cluster configuration
+  echo "om"  # Default OM service name in docker-compose
+}
+
+_download_checkpoint_v1() {
+  _kinit
+  local om_host=$(_get_om_hostname)
+  local expected_result="$1"
+
+  echo "Testing /dbCheckpoint endpoint: client ${client_version} → cluster ${cluster_version}"
+
+  # Add debugging information
+  echo "DEBUG: Using container: ${container}"
+  echo "DEBUG: Using OM host: ${om_host}"
+  
+  # Check if OM is reachable
+  echo "DEBUG: Testing OM connectivity..."
+  execute_command_in_container ${container} curl -v -s --connect-timeout 5 "http://${om_host}:9874/" || echo "DEBUG: Basic OM connectivity failed"
```

Comment:
The updates made to this file should be done using robot, not shell scripts. If these tests fail they will not show up in the test report and specifying whether to fail the whole suite or where to continue from is more complicated.